src/SIM/SIMsolution.C

```diff
@@ -46,9 +46,11 @@ bool SIMsolution::saveSolution (SerializeMap& data,
 {
 #ifdef HAS_CEREAL
   std::ostringstream str;
-  cereal::BinaryOutputArchive archive(str);
-  for (const Vector& sol : this->getSolutions())
-    archive(sol);
+  {
+    cereal::BinaryOutputArchive archive(str);
+    for (const Vector& sol : this->getSolutions())
+      archive(sol);
+  }
```

Comment:
It does not have an explicit `close`method one can use instead?

Reply:
unfortunately no.

Reply:
the alternative is new / delete.